lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/wires/types/JsWiresShape.java

```diff
@@ -0,0 +1,288 @@
+
+                case "?shapeType=BORDER&renderType=FILL":
+                    return shape.getFillColor();
+            }
+        }
+        return null;
+    }
+
+    public Shape getBorderColorShape() {
+        final JsArray<Shape> shapeJsArray = flatShapes();
+
+        for (int i = 0; i < shapeJsArray.length; i++) {
+            final Shape shape = shapeJsArray.getAt(i);
+            final Object userData = shape.getUserData();
+            if (userData == null) {
+                continue;
+            }
+
+            String tag = (String) userData;
+
+            switch (tag) {
+                case "?shapeType=BORDER&renderType=STROKE":
+                    return shape;
+
+                case "?shapeType=BORDER&renderType=FILL":
+                    return shape;
+            }
+        }
+        return null;
+    }
+
+    public void draw() {
+        shape.refresh();
+    }
+
+    public void setBackgroundColor(String backgroundColor) {
+        final JsArray<Shape> shapeJsArray = flatShapes();
+
+        if (hasBackgroundColor()) {
+            for (int i = 0; i < shapeJsArray.length; i++) {
+                final Shape shape = shapeJsArray.getAt(i);
+                final Object userData = shape.getUserData();
+                if (userData == null) {
+                    continue;
+                }
+
+                String tag = (String) userData;
+
+                if (tag.equals("?shapeType=BACKGROUND")) {
+                    shape.setFillColor(backgroundColor);
+                    draw();
+                }
+            }
+        } else {
+            Shape shape = getBorderColorShape();
+            if (shape != null) {
+                shape.setFillColor(backgroundColor);
+                draw();
+            }
+        }
+    }
+
+    public String getBackgroundColor() {
+        final JsArray<Shape> shapeJsArray = flatShapes();
+
+        for (int i = 0; i < shapeJsArray.length; i++) {
+            final Shape shape = shapeJsArray.getAt(i);
+            final Object userData = shape.getUserData();
+            if (userData == null) {
+                continue;
+            }
+
+            String tag = (String) userData;
+            if (tag.equals("?shapeType=BACKGROUND")) {
+                return shape.getFillColor();
+            }
+        }
+
+        return null;
+    }
+
+    public boolean hasBackgroundColor() {
+        final JsArray<Shape> shapeJsArray = flatShapes();
+
+        for (int i = 0; i < shapeJsArray.length; i++) {
+            final Shape shape = shapeJsArray.getAt(i);
+            final Object userData = shape.getUserData();
+            if (userData == null) {
+                continue;
+            }
+
+            String tag = (String) userData;
+            if (tag.equals("?shapeType=BACKGROUND")) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public Point2D getBounds() {
+        double width = this.getBoundingBox().getWidth();
```

Comment:
please check calls to getBoundingBox are "cached", becuase BB calculatiion is expensive, otherwise just create a varialbe on top

Reply:
Hi @romartin fixed

Reply:
Thanks!